app/models/agents/weibo_publish_agent.rb

```diff
@@ -76,6 +88,24 @@ def publish_tweet text
       weibo_client.statuses.update text
     end
 
+    def publish_tweet_with_pic text, pic
+      weibo_client.statuses.upload text, open(pic)
+    end
+
+    def valid_image?(url)
+      begin
+        url = URI.parse(url)
```

Comment:
If we did not check at all, would the Weibo system accept our message and ignore the picture?  If yes, maybe we can skip this check?